Core.UnitTests/Serialization/SerializationEventRaiserTest.cs

```diff
@@ -78,20 +78,15 @@ public void InvokeAttributedMethod_UsesCache ()
       ClassWithDeserializationEvents instance = new ClassWithDeserializationEvents ();
       StreamingContext context = new StreamingContext();
 
-      MockRepository repository = new MockRepository();
-      SerializationEventRaiser eventRaiserMock = repository.StrictMock<SerializationEventRaiser>();
+      var eventRaiserMock = new Mock<SerializationEventRaiser> {CallBase = true};
 
-      eventRaiserMock.InvokeAttributedMethod (instance, typeof (OnDeserializedAttribute), context);
-      LastCall.CallOriginalMethod (OriginalCallOptions.CreateExpectation);
+      eventRaiserMock.Object.InvokeAttributedMethod (instance, typeof (OnDeserializedAttribute), context);
+      eventRaiserMock.Protected().Verify ("FindDeserializationMethodsWithCache", Times.Once(), typeof (ClassWithDeserializationEvents), typeof (OnDeserializedAttribute));
+      eventRaiserMock.Protected().Verify ("FindDeserializationMethodsNoCache", Times.Once(), typeof (ClassWithDeserializationEvents), typeof (OnDeserializedAttribute));
 
-      Expect.Call (PrivateInvoke.InvokeNonPublicMethod (eventRaiserMock, "FindDeserializationMethodsWithCache", typeof (ClassWithDeserializationEvents), typeof (OnDeserializedAttribute)))
-          .Return (new List<MethodInfo>());
-
-      repository.ReplayAll();
-
-      eventRaiserMock.InvokeAttributedMethod (instance, typeof (OnDeserializedAttribute), context);
-
-      repository.VerifyAll();
+      eventRaiserMock.Object.InvokeAttributedMethod (instance, typeof (OnDeserializedAttribute), context);
+      eventRaiserMock.Protected().Verify ("FindDeserializationMethodsWithCache", Times.Exactly (2), typeof (ClassWithDeserializationEvents), typeof (OnDeserializedAttribute));
```

Comment:
okay, this only supports additive constraints? hmm... 

Do we have https://stackoverflow.com/questions/4163679/reset-mock-verification-in-moq
`Moq.MockExtensions.ResetCalls()` and/or `foo.Invocations.Clear()`
available already? If so, can we use this to reset the first recorded state and then only check the second half?